### ISSUE TITLE
Count problematic in pagination

### ISSUE THREAD

**Issue by sorrentinov** [tkaczmarzyk/specification-arg-resolver#51]
Hi,

I see that when I use @Join annotation sometime I've a wrong number of elements in count. 
The reason why this happen depend on the query that has been executed:
Example
 ```
public class Institute {
   @Id 
    private Long id;
    .......
}
```
```
public class Site {
    @Id 
    private Long id;

    @Column("NAME")
    private String name;

   @ManyToOne(optional = false)
    @JoinColumn(name = "ID_INSTITUTION", referencedColumnName = "ID", nullable = false )
    private Institution institution;

    @ManyToOne
    @JoinColumn(name = "ID_SITE_BC_DR", referencedColumnName = "ID", nullable = true)
    private Site siteBcDr;
}
```
```
@Join(path= "institution", alias = "inst"),
@And({
    @Spec(path="inst.id",   params="institutionId", spec=Equal.class),
    @Spec(path="name",                              spec=Like.class),
}) 
public interface SiteFilter extends Specification<Site> {
}
```

When I execute repository.findAll(filter, pageable) where filter is an instance of SiteFilter I see in the log:

```
[EL Fine]: sql: 2019-05-06 14:21:54.617--ClientSession(262708237)--Connection(1154579500)--SELECT DISTINCT t1.ID AS a1, ........FROM INSTITUTION t0, SITE t1 WHERE ((t1.ID_INSTITUTION = ?) AND (t0.ID = t1.ID_INSTITUTION)) LIMIT ?, ?
        bind => [1, 0, 10]
[EL Fine]: sql: 2019-05-06 14:21:54.701--ClientSession(262708237)--Connection(1154579500)--SELECT ID, ...... FROM INSTITUTION WHERE (ID = ?)
        bind => [1]
[EL Fine]: sql: 2019-05-06 14:21:54.784--ClientSession(262708237)--Connection(1154579500)--SELECT COUNT(t0.ID) FROM SITE t0, SITE t2, INSTITUTION t1 WHERE ((t2.ID_INSTITUTION = ?) AND (t1.ID = t2.ID_INSTITUTION))
        bind => [1]
```

So, the query has executed correctly but the count add SITE t2 join hence the count is not correct

Could you verify?

Best Regards
Vincenzo

**Comment by tkaczmarzyk** [comment]
the fix is merged - this should be fixed in v2.2.1 release (planned for this week, stay tuned)

**Comment by tkaczmarzyk** [comment]
Thanks to the great contributors, version 2.2.1 has been just released and deployed to the Maven repo (it might take a few hours until it's visible there).
This should fix the reported issue so I'm closing it (but let me know if you are still experiencing it!)
Sorry for the delay!
